etc/docs/Design.md

```diff
@@ -0,0 +1,100 @@
+# Things to cover
+
+## API
+
+### High level Library API
+
+The library API should provide the following methods to the clients:
+
+* Initialize an executor instance with the following settings
+  * Limits for resources
+    
+
+* Execute a command with the supplied arguments, that returns a pointer to a `Command` object that can be later used to manipulate the underlying process
+
+### GRPC API definition
+
+The GRPC API defines the following methods:
+
+* ExecuteCommand: Starts a command with limited resources in its own namespace with the supplied arguments
+* StopExecution : Stop a certain execution 
+* ListExecutions: List all the executions the connected client has rights to
+* GetExecution: Gets metadata about a certain execution defined by an execution ID
+* GetOutStream: Get a stream of lines of standard output (stdout & stderr) of a certain execution
+* GetOut: Get all lines of standard output (stdout & stderr) that has been written so far by the process
+
+To read about the exact details of the API you can look at [the proto file](api.proto)
+
+### Streaming data
+
+In order to be able to stream and also send the full output of the processes we need to store the output of this in memory for simplicty, in order to be able to keep memory from being overloaded a simple recylcing of these processes will be implemented. See [Tradeoffs](#tradeoffs) section for information about why this solution should not be used in production  
```

Comment:
In favor of simplicity, you don't need to implement any recycling.

Reply:
I add it to the tradeoffs then